### DISCUSSION TITLE
New installer/runner for kohya_ss

### DISCUSSION THREAD

**Discussion by bmaltais** [bmaltais/kohya_ss#715] [Announcements]
Hello everyone,

@jstayco has been been hard at work on a new installer. This will be a significant upgrade from the current solution but will bring unification across all platforms (linux, macos and windows).

Since this will be such a radical change I would like to ask for your contribution in testing and reporting issues with the solution that you might experience/observe.

The current PR for the solution is still not merged yet but it is getting close to something that would be ready for user testing: https://github.com/bmaltais/kohya_ss/pull/680

Once the solution is merged in a beta branch I will let you know and provide some instructions on how to load it and test it.

**Comment by 15ky3** [comment]
I´m here to test 👍 just let me know
Linux user and will test it on vast.ai machines

**Comment by bmaltais** [comment]
The new installer is now merged in the dev branch.

If you want to test it you can do this:

```
git checkout dev
git pull
```

Then run the platform specific setup. 

For Linux/MacOS: `./setup.sh`
For Windows: `.\setup.bat` or `.\setup.ps1`

If you want to go back to the master branch do:

```
git checkout master
git pull
```

Then re-run the setup for the platform to re-install things for master... in doubt, delete the venv and logs folder before running setup.

**Reply by jstayco** [reply]
"If you want to go back to the master branch " it's even easier as you can use the new installer for that as well.
```
launcher.py -d /some/other/folder
```
That's it. It will by default install the stable "master" branch from this repo if --branch and --git-repo aren't filled out. Default repo is this repo. Default branch="master".

And it doesn't necessarily require git to be installed either as it will fall back to attempting to downloading the latest release as a zip file from Github if that fails.

**Comment by 15ky3** [comment]
Installer dont work for Machines from vast.ai. Dont know if it works on "normal" Linux Machines because i dont have a Graphics Card to test this.
Got the following error:

`ERROR: Command '/opt/conda/bin/python3.10 -m pip install --upgrade --use-pep517 --no-warn-script-location /workspace/kohya_ss
 --quiet' returned non-zero exit status 1
ERROR: An error occurred during pip operations: Command '['/opt/conda/bin/python3.10', '-m', 'pip', 'install', '--upgrade', '--use-pep517', '--no-warn-script-location', '/workspace/kohya_ss\n', '--quiet']' returned non-zero exit status 1.`

**Reply by jstayco** [reply]
Could you call setup.sh or launcher.py with `-vvv` and `-l $logDir`? Replace $logDir with a writable folder. You will find a time-stamped log file in there. That should give us a bit more info on the exact failure.

Wonder if it's a normal error or maybe because conda is in that image pip is not?

**Reply by 15ky3** [reply]
Sure, will try it tomorrow, running a training right now.
Wrote a Shell Script to install this on vast.ai, maybe it help someone. Working well with a RTX3090.

Script is here: [klick](https://github.com/15ky3/kohya-vast-installer)

**Reply by jstayco** [reply]
That does help! My next step is try to containerize all of this so everyone has reliable images to pull from, but I don't have the cloud services to test a lot with, so these kinds of scripts are gold. Thank you very much!

**Reply by 15ky3** [reply]
Im new to all this Stable Diffusion stuff, just learning to create LORAs but i have to learn much, doesnt work very well at the moment xD

BLIP Captioning only works with the torchvision Version provided with the setup.sh script, Training works with my Script.
Im here to test :) vast isnt expensive, so i will gladly test your scripts.
Would be interesting if i can use own docker container. Should try this the next days

**Comment by Kadah** [comment]
Install on Ubuntu 22.04, though untested for functionality beyond getting the GUI running (need to get datasets over to that system).

`--torch-version` does nothing, not if it should do anything yet. I've been running Torch2 with the main branch on my main rig for several weeks without issue.

**Reply by jstayco** [reply]
Well, that's embarrassing. I added the torch version support to setup.sh, but forgot to pass it through to launcher.py. Apologies there. I'll get a fix out. For now if you need it? You can just do `./launcher.py -t 2' + whatever options you need. setup.sh is only to install python, tk, and git.

**Comment by tenceoner** [comment]
any word on this guy installing but then pretending to be a potato???got it to the ui but nothing responds at all, when i paste addys into text boxes i get an error but nothing is being printed in console... any hiiints?????i ve sucsessfully isntalled and have been runnning both oooogabooga and automatic 1111 but this ones trying to make me look dumb in front of my friends lol
